mat/dense.go

```diff
@@ -67,6 +67,27 @@ func NewDense(r, c int, data []float64) *Dense {
 	}
 }
 
+// ReuseAs changes the receiver if it IsZero() to be of size r×c.
+//
+// ReuseAs re-uses the backing data slice if it has sufficient capacity,
+// otherwise a new slice is allocated. The data is then zeroed.
+//
+// ReuseAs panics if the receiver is not zero-sized, and panics if
+// the input sizes are less than one. To zero a matrix for re-use,
+// Reset should be used.
+func (m *Dense) ReuseAs(r, c int) {
+	if r <= 0 || c <= 0 {
+		if r == 0 || c == 0 {
+			panic(ErrZeroLength)
+		}
+		panic(ErrNegativeDimension)
+	}
+	if !m.IsZero() {
+		panic("mat: ReuseAs called on non-zero matrix")
```

Comment:
This should be made one of the Error types

Reply:
Yes, with the text "mat: reuse non-zero matrix".

Reply:
Done.